empty interface declaration

The following program `sample.go` triggers a panic:

```go
package main

import "fmt"

var I interface{}

func main() {
    fmt.Printf("%T, %v\n")
}
```

Expected result:
```console
$ go run ./sample.go
<nil>, <nil>
```

Got:
```console
$ yaegi ./sample.go
./sample.go:8:2: panic
panic: reflect: Call using zero Value argument [recovered]
        panic: reflect: Call using zero Value argument

goroutine 1 [running]:
github.com/containous/yaegi/interp.runCfg.func1(0xc0000d3680, 0xc0000dd500)
        /home/marc/go/src/github.com/containous/yaegi/interp/run.go:114 +0x218
panic(0xc9da60, 0xc00013c5a0)
        /usr/lib/go/src/runtime/panic.go:679 +0x1b2
reflect.Value.call(0xce3080, 0xe0d1a0, 0x13, 0xdd1a58, 0x4, 0xc0000d56d0, 0x3, 0x3, 0xd244a0, 0xc00000f340, ...)
        /usr/lib/go/src/reflect/value.go:385 +0x14f8
reflect.Value.Call(0xce3080, 0xe0d1a0, 0x13, 0xc0000d56d0, 0x3, 0x3, 0x51800, 0xc7c800, 0xc00013c590)
        /usr/lib/go/src/reflect/value.go:321 +0xb4
github.com/containous/yaegi/interp.callBin.func6(0xc0000d3680, 0xc000041b80)
        /home/marc/go/src/github.com/containous/yaegi/interp/run.go:930 +0x1a5
github.com/containous/yaegi/interp.runCfg(0xc0000dd500, 0xc0000d3680)
        /home/marc/go/src/github.com/containous/yaegi/interp/run.go:120 +0x7a
github.com/containous/yaegi/interp.(*Interpreter).run(0xc00023fa20, 0xc0000dcd00, 0xc0000d35e0)
        /home/marc/go/src/github.com/containous/yaegi/interp/run.go:98 +0x239
github.com/containous/yaegi/interp.(*Interpreter).Eval(0xc00023fa20, 0xc0000a21e0, 0x5d, 0x25d, 0xc0000a21e0, 0x5d, 0x0, 0x1)
        /home/marc/go/src/github.com/containous/yaegi/interp/interp.go:367 +0x429
main.main()
        /home/marc/go/src/github.com/containous/yaegi/cmd/yaegi/yaegi.go:82 +0x42e
```

Note also that writing `var I interface{} = nil` instead of `var I interface{}` makes the program work.

--- a/_test/interface19.go
+++ b/_test/interface19.go
@@ -1,0 +1,12 @@
+package main
+
+import "fmt"
+
+var I interface{}
+
+func main() {
+	fmt.Printf("%T %v\n", I, I)
+}
+
+// Output:
+// <nil> <nil>

--- a/interp/cfg.go
+++ b/interp/cfg.go
@@ -1439,8 +1439,14 @@ func (interp *Interpreter) cfg(root *node) ([]*node, error) {
 				}
 			}
 			for _, c := range n.child[:l] {
-				index := sc.add(n.typ)
-				sc.sym[c.ident] = &symbol{index: index, kind: varSym, typ: n.typ}
+				var index int
+				if sc.global {
+					// Global object allocation is already performed in GTA.
+					index = sc.sym[c.ident].index
+				} else {
+					index = sc.add(n.typ)
+					sc.sym[c.ident] = &symbol{index: index, kind: varSym, typ: n.typ}
+				}
 				c.typ = n.typ
 				c.findex = index
 			}

--- a/interp/value.go
+++ b/interp/value.go
@@ -180,9 +180,9 @@ func genValueInterface(n *node) func(*frame) reflect.Value {
 	}
 }
 
-func zeroInterfaceValue(interp *Interpreter) reflect.Value {
+func zeroInterfaceValue() reflect.Value {
+	n := &node{kind: basicLit, typ: &itype{cat: nilT, untyped: true}}
 	v := reflect.New(reflect.TypeOf((*interface{})(nil)).Elem()).Elem()
-	n := &node{interp: interp, kind: basicLit, typ: &itype{cat: nilT, untyped: true}}
 	return reflect.ValueOf(valueInterface{n, v})
 }
 
@@ -193,7 +193,7 @@ func genValueInterfaceValue(n *node) func(*frame) reflect.Value {
 		v := value(f)
 		if v.Interface().(valueInterface).node == nil {
 			// Uninitialized interface value, set it to a correct zero value.
-			v.Set(zeroInterfaceValue(n.interp))
+			v.Set(zeroInterfaceValue())
 			v = value(f)
 		}
 		return v.Interface().(valueInterface).value

--- a/interp/value.go
+++ b/interp/value.go
@@ -192,7 +192,7 @@ func genValueInterfaceValue(n *node) func(*frame) reflect.Value {
 	return func(f *frame) reflect.Value {
 		v := value(f)
 		if v.Interface().(valueInterface).node == nil {
-			// Unitialized interface value, set it to a correct zero value.
+			// Uninitialized interface value, set it to a correct zero value.
 			v.Set(zeroInterfaceValue(n.interp))
 			v = value(f)
 		}

--- a/interp/value.go
+++ b/interp/value.go
@@ -180,11 +180,23 @@ func genValueInterface(n *node) func(*frame) reflect.Value {
 	}
 }
 
+func zeroInterfaceValue(interp *Interpreter) reflect.Value {
+	v := reflect.New(reflect.TypeOf((*interface{})(nil)).Elem()).Elem()
+	n := &node{interp: interp, kind: basicLit, typ: &itype{cat: nilT, untyped: true}}
+	return reflect.ValueOf(valueInterface{n, v})
+}
+
 func genValueInterfaceValue(n *node) func(*frame) reflect.Value {
 	value := genValue(n)
 
 	return func(f *frame) reflect.Value {
-		return value(f).Interface().(valueInterface).value
+		v := value(f)
+		if v.Interface().(valueInterface).node == nil {
+			// Unitialized interface value, set it to a correct zero value.
+			v.Set(zeroInterfaceValue(n.interp))
+			v = value(f)
+		}
+		return v.Interface().(valueInterface).value
 	}
 }
 

